File: java/src/main/java/org/example/s3tests/UserData.java
package org.example.s3tests;

public class UserData {
    public String DisplayName ;
    public String UserID      ;
    public String Email       ;
    public String AccessKey   ;
    public String SecretKey   ;
    public String APIName     ;
    public String KMS         ;
 
    public UserData()
    {
        Init();
    }

    public void Init()
    {
        DisplayName = "";
        UserID      = "";
        Email       = "";
        AccessKey   = "";
        SecretKey   = "";
        APIName     = "";
        KMS         = "";
    }

    public boolean IsEmpty()
    {
    	
        if (!DisplayName.isBlank()) return true;
        if (!UserID     .isBlank()) return true;
        if (!Email      .isBlank()) return true;
        if (!AccessKey  .isBlank()) return true;
        if (!SecretKey  .isBlank()) return true;
        if (!APIName    .isBlank()) return true;
        return false;
    }
}

File: python/s3tests/data/user_data.py
"""User credential and grant data."""

from __future__ import annotations

from dataclasses import dataclass, field


@dataclass
class UserData:
    display_name: str = ""
    id: str = ""
    email: str = ""
    access_key: str = ""
    secret_key: str = ""
    kms: str = ""
    x_auth_token: str = ""

    def to_grantee(self) -> dict:
        return {"ID": self.id, "Type": "CanonicalUser"}

    def to_grant(self, permission: str) -> dict:
        return {"Grantee": self.to_grantee(), "Permission": permission}

    def to_owner(self) -> dict:
        return {"ID": self.id, "DisplayName": self.display_name}
